src/features/chains/useChainConfigs.ts

```diff
@@ -0,0 +1,63 @@
+import { useEffect } from 'react';
+import { z } from 'zod';
+
+import { ChainMap, ChainMetadata, ChainMetadataSchema, objMerge } from '@hyperlane-xyz/sdk';
+
+import { useStore } from '../../store';
+import { fromBase64 } from '../../utils/base64';
+import { logger } from '../../utils/logger';
+import { useQueryParam } from '../../utils/queryParams';
+
+import { ChainConfig } from './chainConfig';
+
+const CHAIN_CONFIGS_KEY = 'chains';
+
+const ChainMetadataArraySchema = z.array(ChainMetadataSchema);
+
+// Use the chainConfigs from the store
+export function useChainConfigs() {
+  return useStore((s) => ({
+    chainConfigs: s.chainConfigsV2,
+    setChainConfigs: s.setChainConfigs,
+  }));
+}
+
+// Use the chainConfigs from the store but with any
+// chainConfigs from the query string merged in
+export function useChainConfigsWithQueryParams() {
+  const { chainConfigs: storeConfigs, setChainConfigs } = useChainConfigs();
+  const queryVal = useQueryParam(CHAIN_CONFIGS_KEY);
+
+  useEffect(() => {
+    if (!queryVal) return;
+    const decoded = fromBase64<ChainMetadata[]>(queryVal);
```

Comment:
Is there not some out of the box way to encode the chain config as plaintext query params?

Reply:
The nested structure makes it difficult. It was this or add a 3rd part lib like `qs` everywhere we want to serialize it.